extensions/amp-truncate-text/1.0/amp-truncate-text.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS as COMPONENT_CSS} from './truncate-text.jss';
+import {CSS} from '../../../build/amp-truncate-text-1.0.css';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {TruncateText} from './truncate-text';
+import {dict} from '../../../src/utils/object';
+import {isExperimentOn} from '../../../src/experiments';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-truncate-text';
+const EXPANDED_ATTR = 'i-amphtml-truncate-expanded';
```

Comment:
1. Do we have a default "expanded" state?
2. Why not just give it a public name, e.g. "expanded"?

Reply:
The `0.1` version didn't include a way to set the initial state, and is always collapsed on load. I considered adding a default `expanded` prop, but didn't in this first pass. WDYT?

Reply:
Let's add "expanded" state to tracker.

Reply:
Resolving this thread to consolidate discussion of this state/prop in thread https://github.com/ampproject/amphtml/pull/32068#discussion_r562761052